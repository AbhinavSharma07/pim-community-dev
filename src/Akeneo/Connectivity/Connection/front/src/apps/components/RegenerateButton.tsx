import React, {FC, useContext} from 'react';
import {IconButton} from '../../common';
import {UpdateIcon} from '../../common/icons';
import {TranslateContext} from '../../shared/translate';

interface Props {
    onClick: () => void;
}

export const RegenerateButton: FC<Props> = ({onClick}: Props) => {
    const translate = useContext(TranslateContext);

    return (
        <IconButton
            onClick={onClick}
            title={translate('akeneo_connectivity.connection.edit_app.credentials.action.regenerate')}
        >
            <UpdateIcon />
        </IconButton>
    );
};
