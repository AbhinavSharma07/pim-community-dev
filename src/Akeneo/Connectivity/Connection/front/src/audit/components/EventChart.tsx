import React, {FC, ReactNode, useContext, useEffect, useState} from 'react';
import {VictoryThemeDefinition} from 'victory';
import {AuditEventType} from '../../model/audit-event-type.enum';
import {useNumberFormatter} from '../../shared/formatter/use-number-formatter';
import {useTranslate} from '../../shared/translate';
import {UserContext} from '../../shared/user';
import {useFetchConnectionsAuditData} from '../api-hooks/use-fetch-connections-audit-data';
import {Chart} from './Chart';
import styled from 'styled-components';
import {PropsWithTheme} from '../../common/theme';

type Props = {
    title: ReactNode;
    eventType: AuditEventType;
    theme: VictoryThemeDefinition;
    selectedConnectionCode?: string;
};
type ChartEntry = {
    x: number;
    y: number;
    xLabel: string;
    yLabel: string;
};

const Title = styled.div`
    color: ${({theme}: PropsWithTheme) => theme.color.purple100};
    font-size: ${({theme}: PropsWithTheme) => theme.fontSize.bigger};
    line-height: 44px;
    text-transform: uppercase;
    font-weight: bold;
`;

const EventChartContainer = styled.div`
    width: 49%;
`;

export const EventChart: FC<Props> = ({title, eventType, theme, selectedConnectionCode}: Props) => {
    const connectionsAuditData = useFetchConnectionsAuditData(eventType);
    const [chartData, setChartData] = useState<Array<ChartEntry>>();
    const uiLocale = useContext(UserContext).get('uiLocale');
    const formatNumber = useNumberFormatter();
    const translate = useTranslate();
    useEffect(() => {
        setChartData(undefined);
        if (undefined === selectedConnectionCode || undefined === connectionsAuditData[selectedConnectionCode]) {
            return;
        }

        const selectedConnectionAuditData = connectionsAuditData[selectedConnectionCode].daily;
        const numberOfData = Object.keys(selectedConnectionAuditData).length;
        const chartData = Object.entries(selectedConnectionAuditData).map(
            ([date, value], index): ChartEntry => {
                const xLabel = new Intl.DateTimeFormat(uiLocale.replace('_', '-'), {
                    weekday: 'long',
                    month: 'short',
                    day: 'numeric',
                }).format(new Date(date));

                return {
                    x: index,
                    y: value,
                    xLabel:
                        index + 1 !== numberOfData
                            ? xLabel
                            : translate('akeneo_connectivity.connection.dashboard.charts.legend.today'),
                    yLabel: 0 === index ? '' : formatNumber(value),
                };
            }
        );

        setChartData(chartData);
    }, [uiLocale, formatNumber, translate, connectionsAuditData, selectedConnectionCode]);

    return (
        <>
            {chartData && (
                <EventChartContainer>
                    <Title>{title}</Title>
                    <Chart data={chartData} theme={theme} />
                </EventChartContainer>
            )}
        </>
    );
};
