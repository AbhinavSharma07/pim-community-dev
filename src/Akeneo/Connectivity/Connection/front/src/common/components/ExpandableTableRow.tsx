import React, {Children, FC, ReactNode, useState} from 'react';
import {getColor, Table} from 'akeneo-design-system';
import styled from 'styled-components';

const LargeCell = styled.td.attrs(({colSpan}) => ({colSpan: colSpan}))`
    border-bottom: 1px solid ${getColor('grey', 60)};
`;
const ShowContextContainer = styled.div`
    display: block;
    margin: 0 auto 20px auto;
    width: 70%;
    border: 1px solid ${getColor('grey', 80)};
    background-color: ${getColor('white')};
`;
const ExpandableRow = styled(Table.Row)<{isExpanded: boolean}>`
    ${({isExpanded}) => (isExpanded ? `
    > td {
        border-bottom: none;
    }
    ` : '')};
`;

type Props = {
    contentToExpand: ReactNode;
};

const ExpandableTableRow: FC<Props> = ({contentToExpand, children}) => {
    const [display, setDisplay] = useState(false);
    const handleClick = () => {
        setDisplay(!display);
    };

    return <>
        <ExpandableRow onClick={handleClick} isExpanded={display}>
            {children}
        </ExpandableRow>
        {display &&
            <Table.Row>
                <LargeCell colSpan={Children.count(children)} data-testid='expanded-row-large-cell'>
                    <ShowContextContainer>
                        {contentToExpand}
                    </ShowContextContainer>
                </LargeCell>
            </Table.Row>
        }
    </>;
};

export default ExpandableTableRow;
