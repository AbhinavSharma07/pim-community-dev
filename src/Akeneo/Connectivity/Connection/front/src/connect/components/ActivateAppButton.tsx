import React, {FC} from 'react';
import {useSecurity} from '../../shared/security';
import {useTranslate} from '../../shared/translate';
import {Button} from 'akeneo-design-system';
import {useRouter} from "../../shared/router/use-router";
import {NotificationLevel, useNotify} from "../../shared/notify";

export const ActivateAppButton: FC<{id: string}> = ({id}) => {
    const translate = useTranslate();
    const security = useSecurity();
    const generateUrl = useRouter();
    const notify = useNotify();
    const isAuthorized = !security.isGranted('akeneo_connectivity_connection_manage_apps');

    const handleClick = () => {
        const url = generateUrl('akeneo_connectivity_connection_apps_rest_get_app_activate_url', {
            'id': id,
        });

        fetch(url, {
            method: 'GET',
            headers: [['X-Requested-With', 'XMLHttpRequest']],
        })
            .then(response => response.json())
            .then(response => {
                window.open(response.url);
            })
            .catch(() => {
                notify(NotificationLevel.ERROR, translate('akeneo_connectivity.connection.connect.apps.activate.flash.error'))
            });
    };

    return (
        <Button onClick={handleClick} level='primary' disabled={isAuthorized}>
            {translate('akeneo_connectivity.connection.connect.marketplace.card.connect')}
        </Button>
    );
};
