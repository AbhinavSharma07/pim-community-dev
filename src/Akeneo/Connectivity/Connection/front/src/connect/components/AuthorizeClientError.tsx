import React, {FC} from 'react';
import styled from 'styled-components';
import {ClientErrorIllustration, Link} from 'akeneo-design-system';
import {useTranslate} from '../../shared/translate';

const Container = styled.div`
    text-align: center;
`;

const ErrorTexts = styled.div`
    text-align: center;
    margin-top: 38px;
`;

const ErrorMessage = styled.h3`
    color: ${({theme}) => theme.color.grey140};
    font-size: 28px;
    font-weight: normal;
    margin: 0;
    line-height: 34px;
`;

const SubText = styled.p`
    color: ${({theme}) => theme.color.grey120};
    font-size: ${({theme}) => theme.fontSize.big};
    margin: 10px;
`;

interface Props {
    error: string;
}

export const AuthorizeClientError: FC<Props> = ({error}) => {
    const translate = useTranslate();

    return (
        <Container>
            <ClientErrorIllustration width={420} height={204} />
            <ErrorTexts>
                <ErrorMessage>{translate(error)}</ErrorMessage>
                <SubText>{translate('akeneo_connectivity.connection.connect.apps.authorize.error.sub_text')}</SubText>
                <Link href='https://www.akeneo.com'>
                    {translate('akeneo_connectivity.connection.connect.apps.authorize.error.link_label')}
                </Link>
            </ErrorTexts>
        </Container>
    );
};
