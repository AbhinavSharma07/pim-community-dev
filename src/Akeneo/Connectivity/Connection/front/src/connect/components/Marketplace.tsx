import React, {FC} from 'react';
import MarketplaceHelper from './MarketplaceHelper';
import {AppIllustration, getColor, getFontSize, SectionTitle} from 'akeneo-design-system';
import {Grid as CardGrid, MarketplaceCard} from './MarketplaceCard';
import {Extension, Extensions} from '../../model/extension';
import {useTranslate} from '../../shared/translate';
import styled from 'styled-components';

const EmptyContainer = styled.section`
    text-align: center;
    padding: 40px;
`;

const EmptyMessage = styled.p`
    color: ${getColor('grey', 140)};
    font-size: ${getFontSize('big')};
`;

type Props = {
    extensions: Extensions;
}

export const Marketplace: FC<Props> = ({extensions}) => {
    const translate = useTranslate();
    const extensionList = extensions.extensions.map((extension: Extension) => (
        <MarketplaceCard key={extension.id} extension={extension} />
    ));

    return <>
        <MarketplaceHelper count={extensions.total} />
        <SectionTitle>
            <SectionTitle.Title>
                {translate('akeneo_connectivity.connection.connect.marketplace.extensions.title')}
            </SectionTitle.Title>
            <SectionTitle.Spacer />
            <SectionTitle.Information>
                {translate(
                    'akeneo_connectivity.connection.connect.marketplace.extensions.total',
                    {
                        total: extensions.total.toString(),
                    },
                    extensions.total
                )}
            </SectionTitle.Information>
        </SectionTitle>
        {extensions.total === 0 ? (
            <EmptyContainer>
                <AppIllustration size={128} />
                <EmptyMessage>
                    {translate('akeneo_connectivity.connection.connect.marketplace.extensions.empty')}
                </EmptyMessage>
            </EmptyContainer>
        ) : (
            <CardGrid> {extensionList} </CardGrid>
        )}
    </>;
}
