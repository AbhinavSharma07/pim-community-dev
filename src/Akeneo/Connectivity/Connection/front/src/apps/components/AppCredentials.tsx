import React, {FC, useContext} from 'react';
import {useHistory} from 'react-router';
import {HelperLink, InlineHelper, Section, SmallHelper} from '../../common';
import {AppCredentials as AppCredentialsInterface} from '../../domain/apps/app-credentials.interface';
import {Translate, TranslateContext} from '../../shared/translate';
import {CopiableCredential} from './credentials/CopiableCredential';
import {Credential, CredentialList} from './credentials/Credential';
import {RegenerateButton} from './RegenerateButton';

type Props = {
    code: string;
    label: string;
    credentials: AppCredentialsInterface;
};

export const AppCredentials: FC<Props> = ({code, label, credentials: credentials}: Props) => {
    const translate = useContext(TranslateContext);
    const history = useHistory();

    return (
        <>
            <Section title={<Translate id='akeneo_connectivity.connection.edit_app.credentials.title' />} />
            <div>
                <SmallHelper>
                    <Translate
                        id='akeneo_connectivity.connection.edit_app.credentials.helper.message'
                        placeholders={{label}}
                    />
                    &nbsp;
                    <HelperLink
                        href={translate('akeneo_connectivity.connection.edit_app.credentials.helper.link_url')}
                        target='_blank'
                        rel='noopener noreferrer'
                    >
                        <Translate id='akeneo_connectivity.connection.edit_app.credentials.helper.link' />
                    </HelperLink>
                </SmallHelper>
            </div>

            <CredentialList>
                <CopiableCredential label={translate('akeneo_connectivity.connection.connection.client_id')}>
                    {credentials.clientId}
                </CopiableCredential>
                <CopiableCredential
                    label={translate('akeneo_connectivity.connection.connection.secret')}
                    actions={<RegenerateButton onClick={() => history.push(`/apps/${code}/regenerate-secret`)} />}
                >
                    {credentials.secret}
                </CopiableCredential>
                <CopiableCredential label={translate('akeneo_connectivity.connection.connection.username')}>
                    {credentials.username}
                </CopiableCredential>
                {credentials.password ? (
                    <CopiableCredential
                        label={translate('akeneo_connectivity.connection.connection.password')}
                        helper={
                            <InlineHelper warning>
                                <Translate id='akeneo_connectivity.connection.edit_app.credentials.clear_password_helper.message' />{' '}
                                <a
                                    href={translate(
                                        'akeneo_connectivity.connection.edit_app.credentials.clear_password_helper.link_url'
                                    )}
                                    target='_blank'
                                    rel='noopener noreferrer'
                                >
                                    <Translate id='akeneo_connectivity.connection.edit_app.credentials.clear_password_helper.link' />
                                </a>
                            </InlineHelper>
                        }
                    >
                        {credentials.password}
                    </CopiableCredential>
                ) : (
                    <Credential
                        label={translate('akeneo_connectivity.connection.connection.password')}
                        helper={
                            <InlineHelper info>
                                <Translate id='akeneo_connectivity.connection.edit_app.credentials.password_helper.message' />{' '}
                                <a
                                    href={translate(
                                        'akeneo_connectivity.connection.edit_app.credentials.password_helper.link_url'
                                    )}
                                    target='_blank'
                                    rel='noopener noreferrer'
                                >
                                    <Translate id='akeneo_connectivity.connection.edit_app.credentials.password_helper.link' />
                                </a>
                            </InlineHelper>
                        }
                        actions={<RegenerateButton onClick={() => history.push(`/apps/${code}/regenerate-password`)} />}
                    >
                        <Translate id='akeneo_connectivity.connection.edit_app.credentials.password_placeholder' />
                    </Credential>
                )}
            </CredentialList>
        </>
    );
};
