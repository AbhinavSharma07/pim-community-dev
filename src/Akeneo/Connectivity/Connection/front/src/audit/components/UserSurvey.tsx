import React from 'react';
import styled from 'styled-components';
import {Translate} from '../../shared/translate';
import {PropsWithTheme} from '../../common/theme';
import surveyImageUrl from '../../common/assets/illustrations/user-survey.svg';
import {ApplyButton} from '../../common/components';

const Title = styled.div`
    color: ${({theme}: PropsWithTheme) => theme.color.grey140};
    font-size: 28px;
`;
const Content = styled.div`
    color: ${({theme}: PropsWithTheme) => theme.color.grey120};
    font-size: ${({theme}: PropsWithTheme) => theme.fontSize.big};
    margin: 10px auto 20px;
`;
const Container = styled.div`
    width: 740px;
    margin: 10px auto;
    text-align: center;
`;
const Illustration = styled.img`
    margin: 0 auto;
    width: 128px;
`;

export interface Props {
    title: string;
    content: string;
    buttonLabel: string;
    link: string;
}

export const UserSurvey = ({title, content, buttonLabel, link}: Props) => {
    const handleClick = () => {
        const win = window.open(link, '_blank');
        if (null !== win) {
            win.focus();
        }
    };

    return (
        <Container>
            <Illustration src={surveyImageUrl} alt={'ziggy illustration'} />
            <Title>
                <Translate id={title} />
            </Title>
            <Content>
                <Translate id={content} />
            </Content>
            <ApplyButton onClick={handleClick}>
                <Translate id={buttonLabel} />
            </ApplyButton>
        </Container>
    );
};
