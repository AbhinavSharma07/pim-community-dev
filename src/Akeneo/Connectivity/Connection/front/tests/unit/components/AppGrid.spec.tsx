import * as React from 'react';
import {AppGrid} from '../../../src/apps/components/AppGrid';
import {createWithTheme} from '../../utils/create-with-theme';

describe('AppGrid', () => {
    it('should render', () => {
        const component = createWithTheme(<AppGrid title={'others'} apps={[]} />);

        expect(component.toJSON()).toMatchSnapshot();
    });
});
