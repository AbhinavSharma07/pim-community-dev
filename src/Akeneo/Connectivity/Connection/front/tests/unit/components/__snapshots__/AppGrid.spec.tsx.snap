// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppGrid should render 1`] = `
Array [
  <header
    className="sc-VigVT bqWYVJ"
  >
    <div
      className="sc-jTzLTM gnUsfy"
    >
      others
    </div>
    <div
      className="sc-kEYyzF uXDi"
    >
      akeneo_connectivity.connection.app_count
    </div>
  </header>,
  <div
    className="sc-kkGfuU fGwjgM"
  />,
]
`;
